apm-agent-plugins/apm-aws-sdk/apm-aws-sdk-2-plugin/src/main/java/co/elastic/apm/agent/awssdk/v2/BaseSyncClientHandlerInstrumentation.java

```diff
@@ -47,6 +48,9 @@
 import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
 
 public class BaseSyncClientHandlerInstrumentation extends TracerAwareInstrumentation {
+    //Coretto causes sigsegv crashes when you try to access a throwable if it thinks
+    //it went out of scope, which it seems to for the instrumented throwable access
+    private static Throwable REDACTED = new Throwable("Unable to provide details of the error");
```

Comment:
We are going to send an error with this throwable, which means that the stack trace will be visible. We need to take care of that somehow. Two options I thought of:
1. use a somewhat-useful stack trace: create this throwable lazily in the right spot (meaning- when you use it for the first time) so that it captures a more useful stack trace. Clone the stack trace without its top part that contains our code (for example by traversing it until you reach the AWS SDK package) and use it instead. It doesn't have to be synchronized because we don't mind doing it more than once.
2. don't use a stack trace: use the `Throwable` constructor that doesn't fill the stack trace.

Reply:
We can't do 1 cos that's exactly what can trigger the crash. Although it seems like the first time has low risk, but since this is to avoid the crash, better to skip that. So 2 is the option. Good call, thanks

Reply:
> We can't do 1 cos that's exactly what can trigger the crash

If any exception is unsafe to handle, then this fix is insufficient. We still treat all other exceptions everywhere the same.

Reply:
Yup, I did misunderstand. I think it's just about worth it. Implementation added, please review

Reply:
Actually, the way you implemented it made me think there is small further improvement available within reach - if you add a `@Advice.This thiz` argument to `exitDoExecute()`, then `thiz.getClass().getName()` would give you the **actual** class name, instead of using `AWS_CLASS_NAME` always, and then your `RedactedException` would be a multiton with an instance per instrumented class.

Reply:
another good call, thanks! added now